File: notifications-polyglot/typescript/src/common/api-response.ts
export class ApiResponse<T> {
    success: boolean;
    message: string;
    data?: T;

    static success<T>(data: T): ApiResponse<T> {
        return { success: true, message: 'Success', data };
    }

    static error<T>(message: string): ApiResponse<T> {
        return { success: false, message };
    }
}

File: notifications-polyglot/kotlin/src/main/kotlin/com/example/notification/common/ApiResponse.kt
package com.example.notification.common

data class ApiResponse<T>(
    val success: Boolean,
    val message: String,
    val data: T?
) {
    companion object {
        fun <T> success(data: T): ApiResponse<T> = ApiResponse(true, "Success", data)
        fun <T> error(message: String): ApiResponse<T> = ApiResponse(false, message, null)
    }
}
